npc/re/merchants/enchan_illusion_16_2.txt

```diff
@@ -0,0 +1,250 @@
+//===== rAthena Script =======================================
+//= Episode 16.2 - Illusion Enchants
+//===== Description: =========================================
+//= [Walkthrough Conversion]
+//= Illusion series related merchants and enchanters
+//===== Changelogs: ==========================================
+//= 1.0 Initial release [crazyarashi]
+//= 1.1 Cleanup and improvements [Everade]
+//============================================================
+
+//============================================================
+//= Illusion of Moonlight
+//============================================================
+
+pay_d03_i,160,45,3	script	Gemcutter#ilp20	4_TOWER_17,3,3,{
+	mes "[ Gemcutter ]";
+	mes "Do you have business with me?";
+	next;
+	switch (select("What are you doing here?:Upgrade Weapon.:Upgrade Armor.")) {
+	case 1:
+		mes "[ Gemcutter ]";
+		mes "I came to the ruined village,";
+		mes "looking for some materials. Touched some strange light, and now I'm here.";
+		next;
+		mes "[ Gemcutter ]";
+		mes "I've decided to stick around for a little while. Figured I'd be safe so long as I stay close to this soldier.";
+		mes "I have a proposition for you. I want some materials that can only be found in this place.";
+		next;
+		mes "[ Gemcutter ]";
+		mes "Get them for me, and I'll improve your equipment in exchange. Just so you know, I can only improve certain types.";
+		next;
+		mes "[ Gemcutter ]";
+		mes "If you're interested, we can discuss the details of our bargain.";
+		close;
+	case 2:
+		disable_items;
+		if (checkweight(25271,1) == 0 || (MaxWeight - Weight) < 1000) {
+			mes "- You're carrying too many items to proceed. -";
+			close;
+		}
+		setarray .@reward_id[0],
+			26109,		// Illusion Staff of Bordeaux
+			28725,		// Illusion Moonlight Dagger
+			16063,		// Illusion Long Mace
+			26007;		// Illusion Spectral Spear
+		.@size = getarraysize(.@reward_id);
+
+		.@refine_req = 9;
```

Comment:
The min refine is +7 in your copy, +9 in mine. It seems to be +9 according to several source, has it been changed in KRO ?

Reply:
I've only verificated the item ids and prices, not the refine requirements. So could be an error.